src/domain/mongoose_domain_sql.erl

```diff
@@ -317,43 +327,46 @@ insert_domain_settings(Pool, Domain, HostType) ->
 delete_domain_settings(Pool, Domain) ->
     execute_successfully(Pool, domain_delete_settings, [Domain]).
 
-set_enabled(Domain, Enabled) when is_boolean(Enabled) ->
+set_domain_for_deletion_settings(Pool, Domain) ->
+    ExtStatus = status_to_int(deleting),
+    execute_successfully(Pool, domain_update_settings_status, [ExtStatus, Domain]).
+
+set_status(Domain, Status) ->
```

Comment:
Just a "nice to have": I think it might be good to have specs for functions using the new status.